### PR TITLE
Add Builtins

### PR DESCRIPTION
**Pull request by ArielAbreu**
Add built-in shell functions into runtimejs. Functions are (currently):
  * echo
  * clear
  * shutdown
  * reboot

...and more commands are a work in progress.

### CODE REVIEW

**Comment by iefserge** [comment]
@ArielAbreu Nice! Let's rename `builtins` to `shell` though, so it's clear that it's a shell stuff.

Random thought, not really related to this PR, do you think it makes sense to rename `shutdown` to `poweroff`? The latter sounds better to me.

**Comment by ArielAbreu** [comment]
Yeah, plus it is the Unix name for it.
I'll rename it.

**Comment by ArielAbreu** [comment]
@iefserge Items renamed, waiting for the result of the Travis build.

**Comment by ArielAbreu** [comment]
@iefserge I do have a small question, though. Are isolates still used? Because I get the feeling (from the tests) that they are sort of being moved away from.

**Comment by iefserge** [comment]
Looks great!
Isolates as in current implementation are deprecated and API is going to be removed. However they might be added back later in a form of web workers or something similar.

File: js/index.js [hunk 1]
```diff
@@ -43,6 +43,11 @@ runtime.shell.setCommand('1', function(args, cb) {
   });
 });
 
+// Builtin functions
+require('./builtins/clear')(runtime);
```
Comment (iefserge) [inline js/index.js]: @ArielAbreu  needs to be renamed as well

**Comment by ArielAbreu** [comment]
Sorry, slipped my mind. So, (about the isolates) how do I save a file from my app into the VFS, just for the session? (Since isolates will be removed)

**Comment by ArielAbreu** [comment]
I'm planning to add `ping`, but I need `runtime-node-net` to be automatically added to the initrd created by `runtimeify`.

**Comment by iefserge** [comment]
VFS is not supported too, what are you trying to do?

**Comment by ArielAbreu** [comment]
i want to be able download files and manipulate them.

**Comment by iefserge** [comment]
ah, you'd need to keep them in buffers or process on the fly.

**Comment by ArielAbreu** [comment]
Ok, thanks. BTW, opened an issue on runtimeify about adding runtime-node-net and http-node.

**Comment by kesla** [comment]
Sorry, now this is already merged so I guess I'm a little bit slow on the ball here - but couldn't this be created as a separate module outside of core?

**Comment by piranna** [comment]
> but couldn't this be created as a separate module outside of core?

:+1:

**Comment by ArielAbreu** [comment]
Well, yes. It'd actually be better as an optional module. `runtime-starter-pack`? :smile: Seriously, I think that'd be better. `runtime-shell` or whatever. Naming things is hard.

**Comment by iefserge** [comment]
@kesla @piranna It's going to be moved out later with all the shell stuff, first need to figure out sane interface for it.

**Comment by kesla** [comment]
Perfect, interface first, modularisation second!



—

**Pull request merged**